include/zephyr/net/ieee802154_radio.h

```diff
@@ -1786,6 +1786,29 @@ struct ieee802154_radio_api {
 	 */
 	int (*continuous_carrier)(const struct device *dev);
 
+	/**
```

Comment:
It might be good to surround this with a Kconfig, if it is not normally available during regular device operation.

Reply:
@cfriedt Both `continuous_carrier` and `modulated_carrier` on the nRF platform driver require extra Kconfig `CONFIG_NRF_802154_CARRIER_FUNCTIONS` but I don't see why other platforms would have to follow the same logic. So I would prefer to leave it as is.

Reply:
Why not simply have a generic carrier Kconfig option here and apply the same one in the Nordic driver (or both?)

Additional reasons:

why expose a NULL function pointer to API consumers?

If a device isn't built for test and is never intended to transmit CW, then code won't accidentally compile or call it, or worse, make that functionality available in the field, possibly jamming other devices?

Zephyr should be minimal by default

Reply:
Please take a look at updated version.